lib/rubocop/cop/lint/duplicated_gem.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # A Gem's requirements should be listed only once in a Gemfile.
+      # @example
+      #   # bad
+      #   gem 'rubocop'
+      #   gem 'rubocop'
+      #
+      #   # bad
+      #   group :development do
+      #     gem 'rubocop'
+      #   end
+      #
+      #   group :test do
+      #     gem 'rubocop'
+      #   end
+      #
+      #   # good
+      #   group :development, :test do
+      #     gem 'rubocop'
+      #   end
+      #
+      #   # good
+      #   gem 'rubocop', groups: [:development, :test]
+      class DuplicatedGem < Cop
+        MSG = 'Gem `%s` requirements already given on line %d ' \
+              'of the Gemfile.'.freeze
+
+        def investigate(processed_source)
+          return unless from_gemfile?(processed_source)
+
+          gem_requirements(processed_source.ast)
+            .each_with_object({}) do |node, gem_lines|
+              required_gem = extract_gem_name(node)
+              check_gem(required_gem, node, gem_lines)
+            end
+        end
+
+        private
+
+        def from_gemfile?(processed_source)
+          File.basename(processed_source.path) == 'Gemfile'
+        end
+
+        def gem_requirements(ast)
+          ast.each_descendant.select { |e| gem_method?(e) }
+        end
+
+        def extract_gem_name(gem_method_node)
+          gem_method_node.children[2].children.first
+        end
+
+        def check_gem(gem_name, node, gem_lines)
+          if gem_lines.key?(gem_name)
+            offense(node, gem_name, gem_lines[gem_name])
+          else
+            gem_lines[gem_name] = node.loc.line
+          end
+        end
+
+        def offense(node, gem_name, line_of_first_occurance)
+          line_range = node.loc.column...node.loc.last_column
+
+          add_offense(
+            node,
+            source_range(processed_source.buffer, node.loc.line, line_range),
+            format(MSG, gem_name, line_of_first_occurance)
+          )
+        end
+
+        def_node_matcher :gem_method?, <<-PATTERN
```

Comment:
You can probably just use [`def_node_search`](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/node_pattern.rb#L485) here and drop the `ast.each_descendant` stuff above